### PR TITLE
Cm/connect frontend and backend

### PR DESCRIPTION
**Pull request by cmishra**
Changes:
 - optimized the dockerfile for the frontend so we build locally and put it into the production container (instead of running `npm install` + building without cache in a build container each time)
 - added webapp URL to allowed CORS of the backend

### CODE REVIEW

File: react-task-tracker/Makefile [hunk 1]
```diff
@@ -5,6 +5,8 @@ GCLOUD_DOCKER_NAME_PREFIX=$(GCLOUD_CONTAINER_URL)/$(GCLOUD_PROJ_NAME)
 
 
 build-docker:
+	npm install
```
Comment (Oceanliu) [inline react-task-tracker/Makefile:8]: any reason moving these from docker file to make file?

**Review by Oceanliu — APPROVED** [review]
lets add `package-lock.json` to .gitignore file

File: react-task-tracker/Makefile (hunk 1 above)
Comment (Oceanliu) [reply react-task-tracker/Makefile:8]: oh nvm I saw your comments

Comment (cmishra) [reply react-task-tracker/Makefile:8]: much faster, building/uploading went from a minute to ~10 seconds with this change.

**Comment by cmishra** [comment]
package-lock.json it should be versioned @Oceanliu https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5.

**Comment by Oceanliu** [comment]
ah interesting

**Pull request merged**